File: modules/wem-api/src/main/java/com/enonic/wem/api/content/ContentUpdatedEvent.java
package com.enonic.wem.api.content;

import com.google.common.base.Objects;

import com.enonic.wem.api.event.Event;

public final class ContentUpdatedEvent
    implements Event
{
    private final ContentId contentId;

    public ContentUpdatedEvent( final ContentId contentId )
    {
        this.contentId = contentId;
    }

    public ContentId getContentId()
    {
        return contentId;
    }

    @Override
    public String toString()
    {
        return Objects.toStringHelper( this ).
            add( "contentId", this.contentId ).
            omitNullValues().
            toString();
    }

}

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/content/ContentUpdatedEvent.ts
module api.content {

    export interface ContentUpdatedEventJson {
        contentId: string;
    }

    export class ContentUpdatedEvent extends api.event.Event {

        private contentId: api.content.ContentId;

        constructor(contentId: api.content.ContentId) {
            super();
            this.contentId = contentId;
        }

        public getContentId(): api.content.ContentId {
            return this.contentId;
        }

        static on(handler: (event: ContentUpdatedEvent) => void) {
            api.event.Event.bind(api.ClassHelper.getFullName(this), handler);
        }

        static un(handler?: (event: ContentUpdatedEvent) => void) {
            api.event.Event.unbind(api.ClassHelper.getFullName(this), handler);
        }

        static fromJson(json: ContentUpdatedEventJson): ContentUpdatedEvent {
            return new ContentUpdatedEvent(new ContentId(json.contentId));
        }
    }
}